### PR TITLE
Rename as "ssb-server"

### LINKED ISSUE #378: proposal: rename `scuttlebot` repo / pkg to `ssb-server`
see discussion on ssb [here](https://viewer.scuttlebot.io/%25BIn4rQl8o7PxeolhiZCw77fUN5gGpOv6I3gDXhMfP%2BM%3D.sha256).

also if we do this, maybe we should split out bin script to `ssb-cli`.

### PR DESCRIPTION
**Pull request by christianbundy**
Resolves #378 by replacing all internal references.

cc: @ahdinosaur @staltz @dominictarr

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,33 +47,33 @@ it is recommended to get started with patchwork, and then look into git-ssb and
...
-// create a scuttlebot client using default settings
+// create a ssb-server client using default settings
 // (server at localhost:8080, using key found at ~/.ssb/secret)
-ssbClient(function (err, sbot) {
+ssbClient(function (err, ssb-server) {
   if (err) throw err
 
   // publish a message
-  sbot.publish({ type: 'post', text: 'My First Post!' }, function (err, msg) {
+  ssb-server.publish({ type: 'post', text: 'My First Post!' }, function (err, msg) {
```
Comment (staltz) [inline README.md]: This should be ssbServer. PS: there are other occurrences below

Comment (christianbundy) [reply README.md]: Nice catch, thanks!

**Comment by christianbundy** [comment]
I've just pushed a change that moves `npm-shrinkwrap.json` from the `scuttlebot-release` module into this readme, which I propose we use. The differences between the different types of lockfiles:

- [`package-lock.json`](https://docs.npmjs.com/files/package-lock.json): lockfile for libraries (click the link for more details)
- [`npm-shrinkwrap.json`](https://docs.npmjs.com/files/shrinkwrap.json): the above *plus* it's published to NPM and respected when you run `npm install -g ssb-server`, great for applications

Once this PR is merged I propose that we deprecate the `scuttlebot-release` package and archive the GitHub repository. This gives us the benefits of having a lockfile (:tada:) without the hassle of maintaining two repos and modules for the same software.

**Comment by christianbundy** [comment]
For posterity: `%RZW7keS4V1Fv4AvUaVvdo7A4hgtLZOdoKDWXByALdE8=.sha256`

**Comment by staltz** [comment]
Are you sure that the hyphen will work in the CLI? `ssb-server publish`

And what about starting the server? Should the command be `ssb-server server`? I think this PR needs more careful consideration.

**Comment by christianbundy** [comment]
@staltz 

> Are you sure that the hyphen will work in the CLI?

Yep, this works fine (similar to `apt-get`).

> Should the command be `ssb-server server`? 

I've just pushed a commit renaming this command to `ssb-server start` and fixing all references to the command. :tada:

File: package.json [hunk 2]
```diff
@@ -64,8 +64,7 @@
     "tape": "^4.9.1"
   },
   "bin": {
-    "sbot": "./bin.js",
```
[the remark this answers is not in the document]
Comment (christianbundy) [reply package.json]: I think this is a great recommendation, I'll add a comment below detailing my steps.

**Comment by christianbundy** [comment]
@ahdinosaur 

I've just pushed a commmit to ensure backward-compatibility in two areas:

1. Any time `sbot` is invoked we warn the user about the command rename.
2. Any time `sbot server` or `ssb-server server` invoked we warn the user about the subcommand rename.

Also, thanks for cleaning up those mistaken "ssbServer" bits! Anything else you can think of?

cc: @staltz

**Comment by christianbundy** [comment]
Hey @staltz, it's still showing that you're requesting changes on this PR -- are those good to dismiss, or are there any other changes you'd like to see?

**Comment by christianbundy** [comment]
Thanks, merging now!

**Comment by mmckegg** [comment]
wow, end of an era! Scuttlebot is dead, long live ssb-server! 💥

**Pull request merged**